### PR TITLE
[DNM] Port prost to raft 0.4.3

### PR DESCRIPTION
**Pull request by ice1000**
PTAL @nrc 

This PR adds prost migration and Rust 2018 to the newest 0.4.x branch.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "raft"
-version = "0.4.3"
+version = "0.4.4"
```
Comment (nrc) [inline Cargo.toml]: I think we will keep the same version number and use a Git dep

File: tests/integration_cases/mod.rs [hunk 2]
```diff
@@ -12,7 +12,6 @@
 // limitations under the License.
 
 mod test_raft;
-mod test_raft_flow_control;
```
Comment (nrc) [inline tests/integration_cases/mod.rs]: Why was this removed?

**Comment by siddontang** [comment]
don't do this, as I said before, we should use prost in 0.5, not in 0.4

**Comment by ice1000** [comment]
> don't do this, as I said before, we should use prost in 0.5, not in 0.4

This PR is primarily for reviewing purpose, I'll mention this in the title and label

**Comment by ice1000** [comment]
Oh I can't add labels. Can anyone add a "Do Not Merge" label for me?

File: tests/integration_cases/mod.rs (hunk 2 above)
Comment (ice1000) [reply tests/integration_cases/mod.rs]: It's a mistake. I apologize

**Comment by nrc** [comment]
> don't do this, as I said before, we should use prost in 0.5, not in 0.4

I think we need this PR to continue testing/benchmarking at the least. Once we migrate TiKV to 0.5, then we can use 0.5 + Prost, but for now having 0.4 + Prost is blocking my work in TiKV

**Comment by nrc** [comment]
TO clarify, I mean that we should keep this branch in the raft-rs repo, not that we should merge it to master or to publish it.

**Comment by nrc** [comment]
This is now the Prost branch in this repo.
